libs/langchain-azure-cosmosdb/src/chat_histories.ts

```diff
@@ -0,0 +1,166 @@
+import { Container, CosmosClient, CosmosClientOptions } from "@azure/cosmos";
+import { DefaultAzureCredential, TokenCredential } from "@azure/identity";
+import { BaseListChatMessageHistory } from "@langchain/core/chat_history";
+import {
+  BaseMessage,
+  mapChatMessagesToStoredMessages,
+  mapStoredMessagesToChatMessages,
+} from "@langchain/core/messages";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+
+const USER_AGENT_SUFFIX = "langchainjs-cdbnosql-chathistory-javascript";
+const DEFAULT_DATABASE_NAME = "chatHistoryDB";
+const DEFAULT_CONTAINER_NAME = "chatHistoryContainer";
+
+/**
+ * Type for the input to the `AzureCosmosDBNoSQLChatMessageHistory` constructor.
+ */
+export interface AzureCosmosDBNoSQLChatMessageHistoryInput {
+  sessionId: string;
+  userId: string;
```

Comment:
Could we make the userId param optional here, with a fallback to a default one? (like `anonymous` or whatever)?
This would allow the integration to work in context where no user auth is available. Not a big use case I agree, but other history integrations don't force a userId requirement.

Reply:
I'm adding "anonymous" as default for now. But should it be different for different users (e.g. we could add some random digits after "user" or "anonymous")? Considering we are adding method for clearing all sessions for a user. And if someone would pass "anonymous" in this method, it would clear sessions for other users too.